### ISSUE TITLE
One unit test fails with 4 CPUs, fine with 2 CPUs

### ISSUE THREAD

**Issue by gongsu832** [hyperledger-archives/fabric#1921]
## Description

On a zLinux guest (debian 8) with 4 CPUs, "make unit-test" fails for github.com/hyperledger/fabric/core/system_chaincode:

```
2016/06/20 03:35:24 Load docker HostConfig: %+v &{[] [] []  [] false map[] [] false [] [] [] [] host    { 0} [] { map[]} false []  0 0 0 false 0    0 0 0 []}
2016/06/20 03:35:24 Chaincode support using peerAddress: 0.0.0.0:21726
2016/06/20 03:35:24 Is db path [/var/hyperledger/test/tmpdb/db] empty [false]
Error opening DB IO error: lock /var/hyperledger/test/tmpdb/db/LOCK: Resource temporarily unavailable
--- FAIL: TestExecuteDeploySysChaincode (0.00s)
panic: Could not open openchain db error = [IO error: lock /var/hyperledger/test/tmpdb/db/LOCK: Resource temporarily unavailable] [recovered]
    panic: Could not open openchain db error = [IO error: lock /var/hyperledger/test/tmpdb/db/LOCK: Resource temporarily unavailable]

goroutine 8 [running]:
panic(0x806f68a0, 0xc82021ad50)
    /usr/lib/go-1.6/src/runtime/panic.go:464 +0x400
testing.tRunner.func1(0xc82007a3f0)
    /usr/lib/go-1.6/src/testing/testing.go:467 +0x1ac
panic(0x806f68a0, 0xc82021ad50)
    /usr/lib/go-1.6/src/runtime/panic.go:426 +0x536
github.com/hyperledger/fabric/core/db.GetDBHandle(0x4)
    /opt/openchain/src/github.com/hyperledger/fabric/core/db/db.go:106 +0x2c0
github.com/hyperledger/fabric/core/ledger.fetchBlockchainSizeFromDB(0xc820217110, 0x0, 0x0)
    /opt/openchain/src/github.com/hyperledger/fabric/core/ledger/blockchain.go:258 +0x24
github.com/hyperledger/fabric/core/ledger.newBlockchain(0x1, 0x0, 0x0)
    /opt/openchain/src/github.com/hyperledger/fabric/core/ledger/blockchain.go:49 +0x24
github.com/hyperledger/fabric/core/ledger.GetNewLedger(0xc8201ff7e0, 0x0, 0x0)
    /opt/openchain/src/github.com/hyperledger/fabric/core/ledger/ledger.go:101 +0x24
github.com/hyperledger/fabric/core/ledger.GetLedger.func1()
    /opt/openchain/src/github.com/hyperledger/fabric/core/ledger/ledger.go:94 +0x1e
sync.(*Once).Do(0x80d2b650, 0x80a5afe0)
    /usr/lib/go-1.6/src/sync/once.go:44 +0x184
github.com/hyperledger/fabric/core/ledger.GetLedger(0x0, 0x0, 0x0)
    /opt/openchain/src/github.com/hyperledger/fabric/core/ledger/ledger.go:95 +0x3c
github.com/hyperledger/fabric/core/chaincode.Execute(0x3ff9973a538, 0xc82000cea0, 0xc8200e3700, 0xc8200ea460, 0x0, 0x0, 0x0, 0xc8202170e0, 0x0, 0x0)
    /opt/openchain/src/github.com/hyperledger/fabric/core/chaincode/exectransaction.go:36 +0x36
github.com/hyperledger/fabric/core/system_chaincode/api.deploySysCC(0x3ff9973a538, 0xc82000cea0, 0xc8201ab420, 0x0, 0x0)
    /opt/openchain/src/github.com/hyperledger/fabric/core/system_chaincode/api/sysccapi.go:108 +0x4f0
github.com/hyperledger/fabric/core/system_chaincode/api.RegisterSysCC(0xc82017a870, 0x0, 0x0)
    /opt/openchain/src/github.com/hyperledger/fabric/core/system_chaincode/api/sysccapi.go:76 +0xb7e
github.com/hyperledger/fabric/core/system_chaincode.RegisterSysCCs()
    github.com/hyperledger/fabric/core/system_chaincode/_test/_obj_test/importsysccs.go:16 +0x88
github.com/hyperledger/fabric/core/system_chaincode.TestExecuteDeploySysChaincode(0xc82007a3f0)
    /opt/openchain/src/github.com/hyperledger/fabric/core/system_chaincode/systemchaincode_test.go:114 +0x58a
testing.tRunner(0xc82007a3f0, 0x80cd8aa0)
    /usr/lib/go-1.6/src/testing/testing.go:473 +0xe2
created by testing.RunTests
    /usr/lib/go-1.6/src/testing/testing.go:582 +0x6c4
FAIL    github.com/hyperledger/fabric/core/system_chaincode 0.031s

```

Running "make unit-test" with only 2 CPUs (after taking 2 CPUs offline) works fine. Running the single package test with "go test -cover github.com/hyperledger/fabric/core/system_chaincode" also works fine.
## Describe How to Reproduce

make unit-test with 4 CPUs.

**Comment by gongsu832** [comment]
Forgot to mention the code version:

```
commit 35326c25f99b038286a58330fdef87d23fe5f473
Merge: 0d05bf3 a626e43
Author: Binh Q Nguyen <binhn@us.ibm.com>
Date:   Sat Jun 18 08:57:17 2016 -0400

    Merge pull request #1877 from jyellick/keep-state-if-can-execute

    Stabilize PBFT under stress with periodic viewchange
```

**Comment by tuand27613** [comment]
@gongsu832  can you check with a post PR #1926 build ?

**Comment by gongsu832** [comment]
@tuand27613 I can confirm the latest commit 72a7cbf9d3f49ee79d71c494f8aef916b7376251 resolves the issue. Tried on two zLinux VM guests with 4 CPUs, one running Debian 8 and the other running RHEL 7.2. "make unit-test" now works on both guests.

**Comment by srderson** [comment]
Thanks, also fixed via PR #1930. We don't run tests in parallel now.
